FogCreek/Glitch-Community
# Profuse tea: AB testing with a very basic test on the secret page

## Links
https://profuse-tea.glitch.me/secret
https://github.com/FogCreek/Glitch-Community-Work/issues/303
https://github.com/FogCreek/Glitch-Community-Work/issues/304

## GIF/Screenshots:
![image](https://user-images.githubusercontent.com/8932174/63463509-4d3d1f80-c42b-11e9-915c-60c79e9994c9.png)

## Changes:
- Added a basic a/b testing framework on the server. Each test has a unique name and a set of buckets, each with a unique name, a weight, and a value
- The server reads the ab-tests cookie and randomly picks groups for any tests that aren't already assigned
- The tests are exposed in react as testname: groupvalue, so groups can store flags or whatever is needed for the test
- There's a really basic test defined, with 10% good job, 10% nice one, and 80% too bad. It's visible on the secret page, along with a button to wipe the cookie so you can try again and see what group you end up in

## How To Test:
Go to the secret page and try refreshing the page a bunch of times. You should always stay in the same group after a refresh. If you click the button, you should get a new group (in general, it isn't a guarantee)

## Feedback I'm looking for:
Group values are pretty open, they could store alternate text for an element or styling. Maybe it should only be used as flags, and keep the output in the react code?

## Things left to do before deploying:
Maybe remove the example test, or maybe leave it in to see how users get assigned
